source/common/stats/stats_impl.cc

```diff
@@ -153,14 +136,9 @@ bool TagExtractorImpl::extractTag(const std::string& stat_name, std::vector<Tag>
 }
 
 RawStatData* HeapRawStatDataAllocator::alloc(const std::string& name) {
-  RawStatData* data = static_cast<RawStatData*>(::calloc(RawStatData::size(), 1));
+  RawStatData* data = static_cast<RawStatData*>(::calloc(RawStatData::sizeGivenName(name), 1));
```

Comment:
If this no longer truncates, the names of the stats that overflow into the heap allocator in the hot restart case will no longer be truncated.  This means that the overflowed names will be inconsistent with those that were allocated in the block allocator.  Can we make sure that we truncate in that case to avoid the inconsistency?

Reply:
This sounds good. Rather than propagate a truncation throughout `ThreadLocalStore`, I've put the `.substr()` fn inline where `make_stat` calls the `heap_allocator` in `thread_local_store.cc.